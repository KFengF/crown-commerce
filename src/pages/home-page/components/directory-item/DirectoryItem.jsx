import React from 'react';
import { withRouter } from 'react-router-dom';
import './DirectoryItem.scss';

const DirectoryItem = ({ title, imageUrl, size, linkUrl, history, match }) => (
    <div 
        className={ `directory-item ${ size }` }
        onClick={ () => history.push(`${ match.url }${ linkUrl }`) }
    >
        <div className="background" style={{ backgroundImage: `url(${ imageUrl })` }} />
        <div className="content">
            <h1 className="title">{ title }</h1>
            <span className="subtitle">SHOP NOW</span>
        </div>
    </div>
)

export default withRouter(DirectoryItem);
//withRouter para que tenga los props history y match