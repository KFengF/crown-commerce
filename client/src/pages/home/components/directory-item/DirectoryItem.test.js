import React from 'react';
import { shallow } from 'enzyme';
import { DirectoryItem } from './DirectoryItem.component';

describe('DirectoryItem component', () => {
    let wrapper;
    let mockTitle;
    let mockSize;
    let mockLinkUrl;
    let mockHistory;
    let mockMatch;

    beforeEach(() => {
        mockTitle = 'hats';
        mockSize = 'large';
        mockLinkUrl = 'shop/hats';
        mockHistory = { push: jest.fn() };
        mockMatch = { url: '/' };
        const mockProps = {
            title: mockTitle,
            imageUrl: 'https://i.ibb.co/cvpntL1/hats.png',
            size: mockSize,
            linkUrl: mockLinkUrl,
            history: mockHistory,
            match: mockMatch
        }
        wrapper = shallow(<DirectoryItem { ...mockProps } />);
    });

    it('should render DirectoryItem component', () => {
        expect(wrapper).toMatchSnapshot();
    });

    describe('DirectoryItemDiv', () => {
        it('should call history.push when clicked in DirectoryItemDiv component', () => {
            wrapper.find('DirectoryItemDiv').simulate('click');
            expect(mockHistory.push).toHaveBeenCalled();
        });

        it('should be the prop size with value large in DirectoryItemDiv component', () => {
            expect(wrapper.find('DirectoryItemDiv').prop('size')).toBe(mockSize);
        });
    })
    
    describe('imageUrl', () => {
        it(`should be the prop style with value {"backgroundImage": "url(https://i.ibb.co/cvpntL1/hats.png)"} 
            in className background`, () => {
                expect(wrapper.find('.background').prop('style'))
                    .toStrictEqual({"backgroundImage": "url(https://i.ibb.co/cvpntL1/hats.png)"});
            }
        );
    });

    describe('title', () => {
        it('should be the text hats in TitleH2 component', () => {
            expect(wrapper.find('TitleH2').text()).toBe(mockTitle);
        });
    });
});