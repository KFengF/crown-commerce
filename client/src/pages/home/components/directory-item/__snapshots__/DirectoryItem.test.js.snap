// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DirectoryItem component should render DirectoryItem component 1`] = `ShallowWrapper {}`;
